tests/node_eviction_test.go

```diff
@@ -0,0 +1,271 @@
+package tests_test
+
+import (
+	"flag"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"kubevirt.io/kubevirt-ansible/tests"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Node Eviction", func() {
+
+	flag.Parse()
+
+	var vmi tests.VmManifest
+
+	ns := ktests.NamespaceTestDefault
+
+	JustBeforeEach(func() {
+		tests.DownloadFromUrl(vmi.URL, vmi.FilePath)
+		tests.ReplaceStringInFile(vmi.FilePath, tests.CirrosImageDev, tests.CirrosImageTest, -1)
+	})
+
+	BeforeEach(func() {
+		ktests.BeforeTestCleanup()
+
+		args := []string{"get", "node", "--selector=node-role.kubernetes.io/compute=true", "--output=jsonpath={.items..metadata.name}"}
+		output, err := ktests.RunOcCommand(args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		var count int
+		nodes := strings.Fields(output)
+		for _, node := range nodes {
+			args := []string{"get", "node", node}
+			output, err := ktests.RunOcCommand(args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			if !strings.Contains(output, "SchedulingDisabled") && !strings.Contains(output, "NotReady") {
+				count++
+			}
+		}
+		if count < 2 {
+			Skip("not enough schedulable nodes for this testing")
+		}
+
+	})
+
+	AfterEach(func() {
+		// nodes can be in drained status after some tests is failed, ensure uncordoning them after each tests.
+		uncordonNodes()
+	})
+
+	Describe("VirtualMachineInstance Evictions", func() {
+		BeforeEach(func() {
+			vmi = tests.VmManifest{
+				Name:     "vmi-ephemeral",
+				FilePath: "/tmp/vmi-ephemeral.yaml",
+				URL:      "https://raw.githubusercontent.com/kubevirt/kubevirt/master/cluster/examples/vmi-ephemeral.yaml",
```

Comment:
Do not use this `https://raw.githubusercontent.com/kubevirt/kubevirt/master/cluster/examples/vmi-ephemeral.yaml`
please bump your manifest to test/manifest, it's a decision being made and we currently trying.

Reply:
Put the manifest to test/manifest is a bad idea to me:
1. Once these manifests are updated in upstream, you need to update them accordingly. Which means extra maintenance.
2. I don't see any problem to use the manifests from kubevirt/kubevirt, and the advantage to use them is it's a testing as well, which ensure latest manifests from upstream are working.

I don't want to put it to local if there is no good reasons.

Reply:
There always have to have extra maintenance, in our convention docs you can check out why me and Nelly and Lukas agaist that.

>I don't want to put it to local if there is no good reasons.

Putting local or create our internal repo is more manageable for us, what are you gonna do if not doing this way? you push your test file into `kubevirt/kubevirt` directly and hoping them will accept your PR only for QE to test?

Reply:
It's unnecessary extra maintenance, read through the document and I still cannot find a reason why we shouldn't do it in this way, which I believe is more suitable than copy them into tests/manifests.

Reply:
>im against that. think about us as gate keepers, if we just use what developers have, we may miss issues - documentation issues is the easiest example for what we may miss

>also, i believe we will need many more variations than what devs have, and in this case we will end up using 2 repos for manifests